rules/internal/testutils/parse.go

```diff
@@ -80,3 +102,69 @@ func ParseProto3Tmpl(t *testing.T, src string, data interface{}) *desc.FileDescr
 	// Parse the proto as a string.
 	return ParseProto3String(t, protoBytes.String())
 }
+
+// decompile reads compiled Go protos and returns a map of strings
+// corresponding to the original protos.
+//
+// This is used to handle google.api common protos, which some tests need
+// to use.
+func decompile(filenames ...string) map[string]string {
```

Comment:
As protoparser does not support compiled descriptors as dependency, the process of descriptor -> source file -> descriptor is a bummer. I filed a [ticket](https://github.com/jhump/protoreflect/issues/279) in protoreflect two weeks ago, but got no response.

Reply:
Yep, I had the same thought. I may file a pull request to protoreflect in the future, but expect that to take longer. Nothing stops us from altering the implementation then, though.